File: packages/app/src/services/software/definitions/githubDesktop.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'github-desktop';
	public name = 'GitHub Desktop';
	public category = [SoftwareCategory.Development];
	public downloadName = 'GitHubDesktopSetup-x64.exe';
	public icon = 'github-desktop.png';
	public homepage = 'https://desktop.github.com';

	public async resolveDownloadUrl() {
		return ok('https://central.github.com/deployments/desktop/desktop/latest/win32');
	}
}

File: src/lib/software/github_desktop.py
from src.lib.software import BaseSoftware

class GitHubDesktop(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'GitHubDesktopSetup-x64.exe'
        self.silent_install_args = ['--silent']

    def resolve_download_url(self):
        self.url_resolved.emit('https://central.github.com/deployments/desktop/desktop/latest/win32')
